pdf-2/src/main/java/com/baeldung/exceltopdf/ExcelToPDFConverter.java

```diff
@@ -0,0 +1,167 @@
+        FileInputStream inputStream = new FileInputStream(excelFilePath);
+        XSSFWorkbook workbook = new XSSFWorkbook(inputStream);
+        inputStream.close();
+        return workbook;
+    }
+    private static Document createPDFDocument(String pdfFilePath) throws IOException, DocumentException {
+        Document document = new Document();
+        PdfWriter.getInstance(document, new FileOutputStream(pdfFilePath));
+        document.open();
+        return document;
+    }
+    public static void convertExcelToPDF(String excelFilePath, String pdfFilePath) throws IOException, DocumentException {
+        XSSFWorkbook workbook = readExcelFile(excelFilePath);
+        Document document = createPDFDocument(pdfFilePath);
+
+        for (int i = 0; i < workbook.getNumberOfSheets(); i++) {
+            XSSFSheet worksheet = workbook.getSheetAt(i);
+
+            // Add header with sheet name as title
+            Paragraph title = new Paragraph(worksheet.getSheetName(), new Font(Font.FontFamily.HELVETICA, 18, Font.BOLD));
+            title.setAlignment(Element.ALIGN_CENTER);
+            document.add(title);
+
+            createAndAddTable(worksheet, document);
+            // Add a new page for each sheet (except the last one)
+            if (i < workbook.getNumberOfSheets() - 1) {
+                document.newPage();
+            }
+        }
+
+        document.close();
+        workbook.close();
+    }
+
+    private static void createAndAddTable(XSSFSheet worksheet, Document document) throws DocumentException, IOException {
+        PdfPTable table = new PdfPTable(worksheet.getRow(0).getPhysicalNumberOfCells());
+        table.setWidthPercentage(100);
+        addTableHeader(worksheet, table);
+        addTableData(worksheet, table);
+        document.add(table);
+    }
+
+    private static void addTableHeader(XSSFSheet worksheet, PdfPTable table) throws DocumentException, IOException {
+        Row headerRow = worksheet.getRow(0);
+        for (int i = 0; i < headerRow.getPhysicalNumberOfCells(); i++) {
+            Cell cell = headerRow.getCell(i);
+            String headerText = cell.getCellType() == CellType.STRING ? cell.getStringCellValue() : String.valueOf(cell.getNumericCellValue());
+            PdfPCell headerCell = new PdfPCell(new Phrase(headerText, getCellStyle(cell)));
+            headerCell.setHorizontalAlignment(Element.ALIGN_CENTER);
+            table.addCell(headerCell);
+        }
+    }
+
+    private static void addTableData(XSSFSheet worksheet, PdfPTable table) throws DocumentException, IOException {
+        Iterator<Row> rowIterator = worksheet.iterator();
+        while (rowIterator.hasNext()) {
+            Row row = rowIterator.next();
+            if (row.getRowNum() == 0) {
+                continue;
+            }
+            for (int i = 0; i < row.getPhysicalNumberOfCells(); i++) {
+                Cell cell = row.getCell(i);
+                String cellValue;
+                if (cell != null) {
```

Comment:
Can this be null if the for loop has the condition `i < row.getPhysicalNumberOfCells()`?

Reply:
You're correct, the cell variable within the loop shouldn't be null if the condition i < row.getPhysicalNumberOfCells() is met. I'll remove the unnecessary null check.